modules/util/source-formatter/src/main/java/com/liferay/source/formatter/check/UpgradeJavaCheck.java

```diff
@@ -54,15 +55,31 @@ private String _fixImports(JavaClass javaClass, String content)
 			String newImportName = importsMap.get(importName);
 
 			if (newImportName != null) {
-				content = StringUtil.replace(
-					content, StringBundler.concat("import ", importName, ";"),
-					StringBundler.concat("import ", newImportName, ";"));
+				String className = _getClassNameByImport(importName);
+				String newClassName = _getClassNameByImport(newImportName);
+
+				if (className.equals(newClassName)) {
+					content = _replaceImport(
+						content, importName, newImportName);
+				}
+				else {
+					content = _replaceImport(
+						content, importName, newImportName);
+
+					content = _replaceImpl(content, className, newClassName);
+				}
 			}
 		}
 
 		return content;
 	}
 
+	private String _getClassNameByImport(String importName) {
```

Comment:
You could remove this method and use `SourceFormatterUtil.getSimpleName()` instead.